client.go

```diff
@@ -164,8 +164,8 @@ func (c *Client) Enroll(ctx context.Context, logger logrus.FieldLogger, code str
 	}
 
 	// Check for any errors returned by the API
-	if err := r.Errors.ToError(); err != nil {
-		return nil, nil, nil, nil, &APIError{e: fmt.Errorf("unexpected error during enrollment: %v", err), ReqID: reqID}
```

Comment:
Looks like we're no longer capturing the request ID. We log this in mobile_nebula. I think we either need to come up with a solution to continue returning it in APIErrors, or maybe add it to every element in the []APIError (and add it to APIError), .... or just take the easier approach of returning a specific code here (ErrInvalidCode), which will also mean that callers don't all have to parse this themselves.

Reply:
Wouldn't that lose information if the api server were to return an error other than "invalid code"?

Reply:
How do you think we might be able to avoid that, and still return an explicit error for invalid codes here? How would the caller do it?

Reply:
I don't think the server ever returns invalid code along with an additional error. If it did, the caller might want to know that information. So I think what I would do is only return `ErrInvalidCode` if that's the only error. If there are multiple errors, leave the return as it was before.

Reply:
<img width="430" alt="image" src="https://github.com/user-attachments/assets/179bc2ee-9bea-4de2-bbeb-04779b36de3a" />

LGTM